packages/flutter_tools/lib/src/commands/widget_preview.dart

```diff
@@ -125,51 +158,235 @@ final class WidgetPreviewStartCommand extends WidgetPreviewSubCommandBase with C
     final FlutterProject rootProject = getRootProject();
     final Directory widgetPreviewScaffold = rootProject.widgetPreviewScaffold;
 
     // Check to see if a preview scaffold has already been generated. If not,
     // generate one.
-    if (!widgetPreviewScaffold.existsSync()) {
+    final bool generateScaffoldProject = !widgetPreviewScaffold.existsSync();
+    widgetPreviewScaffold.createSync();
+
+    if (generateScaffoldProject) {
       logger.printStatus('Creating widget preview scaffolding at: ${widgetPreviewScaffold.path}');
       await generateApp(
-        <String>['widget_preview_scaffold'],
+        <String>['app', kWidgetPreviewScaffoldName],
         widgetPreviewScaffold,
         createTemplateContext(
           organization: 'flutter',
-          projectName: 'widget_preview_scaffold',
+          projectName: kWidgetPreviewScaffoldName,
           titleCaseProjectName: 'Widget Preview Scaffold',
           flutterRoot: Cache.flutterRoot!,
           dartSdkVersionBounds: '^${cache.dartSdkBuild}',
-          linux: const LocalPlatform().isLinux,
-          macos: const LocalPlatform().isMacOS,
-          windows: const LocalPlatform().isWindows,
+          linux: platform.isLinux,
+          macos: platform.isMacOS,
+          windows: platform.isWindows,
         ),
         overwrite: true,
         generateMetadata: false,
       );
-      await _populatePreviewPubspec(rootProject: rootProject);
+
+      // WARNING: this access of widgetPreviewScaffoldProject needs to happen after we generate the
+      // scaffold project as invoking the getter triggers lazy initialization of the preview scaffold's
+      // FlutterManifest before the scaffold project's pubspec has been generated.
+      // TODO(bkonyi): add logic to rebuild after SDK updates
+      await initialBuild(widgetPreviewScaffoldProject: rootProject.widgetPreviewScaffoldProject);
     }
 
-    // WARNING: this needs to happen after we generate the scaffold project as invoking the
-    // widgetPreviewScaffoldProject getter triggers lazy initialization of the preview scaffold's
-    // FlutterManifest before the scaffold project's pubspec has been generated.
     _previewCodeGenerator = PreviewCodeGenerator(
       widgetPreviewScaffoldProject: rootProject.widgetPreviewScaffoldProject,
       fs: fs,
     );
 
+    // TODO(matanlurey): Remove this comment once flutter_gen is removed.
+    //
+    // Tracking removal: https://github.com/flutter/flutter/issues/102983.
+    //
+    // Populate the pubspec after the initial build to avoid blowing away the package_config.json
+    // which may have manual changes for flutter_gen support.
+    await _populatePreviewPubspec(rootProject: rootProject);
+
     final PreviewMapping initialPreviews = await _previewDetector.initialize(rootProject.directory);
     _previewCodeGenerator.populatePreviewsInGeneratedPreviewScaffold(initialPreviews);
 
+    if (launchPreviewer) {
```

Comment:
Remind me, is the "previewer" a separate standalone window (or IDE-embedded view) that preview widgets would appear in?

If so, what does it mean to launch without a previewer? Would it launch onto a device?

Reply:
That's correct. The previewer (currently) runs in a standalone Flutter desktop application.

This option is mostly for testing and basically does all the work to initialize the scaffold except actually launching it.